internal/collector/recon/sharding.go

```diff
@@ -0,0 +1,608 @@
+	t.containerShardingScannedMaxTime.Describe(ch)
+	t.containerShardingScannedMinTime.Describe(ch)
+	t.containerShardingScannedSuccess.Describe(ch)
+
+	t.containerShardingVisitedAttempted.Describe(ch)
+	t.containerShardingVisitedCompleted.Describe(ch)
+	t.containerShardingVisitedFailure.Describe(ch)
+	t.containerShardingVisitedSkipped.Describe(ch)
+	t.containerShardingVisitedSuccess.Describe(ch)
+
+	t.containerShardingInProgressActive.Describe(ch)
+	t.containerShardingInProgressCleaved.Describe(ch)
+	t.containerShardingInProgressCreated.Describe(ch)
+	t.containerShardingInProgressError.Describe(ch)
+	t.containerShardingInProgressFound.Describe(ch)
+	t.containerShardingInProgressObjectcount.Describe(ch)
+
+	t.containerShardingCandidatesFound.Describe(ch)
+	t.containerShardingCandidatesObjectCount.Describe(ch)
+
+}
+
+// CollectMetrics implements the collector.Task interface.
+func (t *ShardingTask) CollectMetrics(ch chan<- prometheus.Metric) {
+	t.containerShardingAuditRootAttempted.Collect(ch)
+	t.containerShardingAuditRootFailure.Collect(ch)
+	t.containerShardingAuditRootHasOverlap.Collect(ch)
+	t.containerShardingAuditRootNumOverlap.Collect(ch)
+	t.containerShardingAuditRootSuccess.Collect(ch)
+
+	t.containerShardingAuditShardAttempted.Collect(ch)
+	t.containerShardingAuditShardFailure.Collect(ch)
+	t.containerShardingAuditShardSuccess.Collect(ch)
+
+	t.containerShardingCleavedAttempted.Collect(ch)
+	t.containerShardingCleavedFailure.Collect(ch)
+	t.containerShardingCleavedMaxTime.Collect(ch)
+	t.containerShardingCleavedMinTime.Collect(ch)
+	t.containerShardingCleavedSuccess.Collect(ch)
+
+	t.containerShardingCreatedAttempted.Collect(ch)
+	t.containerShardingCreatedFailure.Collect(ch)
+	t.containerShardingCreatedSuccess.Collect(ch)
+
+	t.containerShardingMisplacedAttempted.Collect(ch)
+	t.containerShardingMisplacedFailure.Collect(ch)
+	t.containerShardingMisplacedFound.Collect(ch)
+	t.containerShardingMisplacedPlaced.Collect(ch)
+	t.containerShardingMisplacedSuccess.Collect(ch)
+	t.containerShardingMisplacedUnplaced.Collect(ch)
+
+	t.containerShardingScannedAttempted.Collect(ch)
+	t.containerShardingScannedFailure.Collect(ch)
+	t.containerShardingScannedMaxTime.Collect(ch)
+	t.containerShardingScannedMinTime.Collect(ch)
+	t.containerShardingScannedSuccess.Collect(ch)
+
+	t.containerShardingVisitedAttempted.Collect(ch)
+	t.containerShardingVisitedCompleted.Collect(ch)
+	t.containerShardingVisitedFailure.Collect(ch)
+	t.containerShardingVisitedSkipped.Collect(ch)
+	t.containerShardingVisitedSuccess.Collect(ch)
+
+	t.containerShardingInProgressActive.Collect(ch)
+	t.containerShardingInProgressCleaved.Collect(ch)
+	t.containerShardingInProgressCreated.Collect(ch)
+	t.containerShardingInProgressError.Collect(ch)
+	t.containerShardingInProgressFound.Collect(ch)
+	t.containerShardingInProgressObjectcount.Collect(ch)
+
+	t.containerShardingCandidatesFound.Collect(ch)
+	t.containerShardingCandidatesObjectCount.Collect(ch)
+}
+
+type AuditRoot struct {
```

Comment:
These type names should indicate that they are related to sharding, otherwise it will be confusing to look at them in godoc. Same for the type declarations below.

Reply:
Done